### PR TITLE
Fixes uncuttable stone (#293)

### PR DESCRIPTION
**Pull request by andreas-eberle**
This PR fixes

* never dissapearing and blocking fully cut off stone which appeared if the loaded map contained fully cut off stone.
* corrected positioning of map object used for fully cut off stone
* implemented rendering for that map object type
* fixed removal of map objects when the ground is flattened for a building.

### CODE REVIEW

**Comment by andreas-eberle** [comment]
Also fixes #219.

**Comment by andreas-eberle** [comment]
@michaelzangl: Can you take a look at the changes?

File: jsettlers.common/src/main/java/jsettlers/common/mapobject/EMapObjectType.java [hunk 1]
```diff
@@ -146,7 +150,21 @@
 	public static final EMapObjectType[] VALUES = EMapObjectType.values();
 	public final byte ordinal;
 
-	private EMapObjectType() {
+	public static final Set<EMapObjectType> TO_BE_REMOVED_WHEN_FLATTENED = new HashSet<>(Arrays.asList(
```
Comment (michaelzangl) [inline jsettlers.common/src/main/java/jsettlers/common/mapobject/EMapObjectType.java]: Use EnumSet.of(...)

File: jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/MapObjectDrawer.java [hunk 2]
```diff
@@ -338,10 +342,6 @@ private void drawObject(int x, int y, IMapObject object, float color) {
 			drawByProgress(x, y, 13, 42, progress, color);
 			break;
 
-		case WINE:
```
Comment (michaelzangl) [inline jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/MapObjectDrawer.java]: Where did this go?

**Comment by michaelzangl** [comment]
The rest looks fine to me.

File: jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/MapObjectDrawer.java (hunk 2 above)
Comment (andreas-eberle) [inline jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/MapObjectDrawer.java]: It was used a long time ago. But we then introduced WINE_GROWING, WINE_HARVESTABLE, WINE_DEAD (like it is also done for CROP_*)

File: jsettlers.common/src/main/java/jsettlers/common/mapobject/EMapObjectType.java (hunk 1 above)
Comment (andreas-eberle) [inline jsettlers.common/src/main/java/jsettlers/common/mapobject/EMapObjectType.java]: thx.

**Pull request merged**